libcloudforensics/providers/azure/internal/account.py

```diff
@@ -381,6 +387,316 @@ def CreateDiskFromSnapshotURI(
+    """
+    vm_sizes = self.ListInstanceTypes()
+    for size in vm_sizes:
+      if size['CPU'] == cpu_cores and size['Memory'] == memory_in_mb:
+        instance_type = size['Name']  # type: str
+        return instance_type
+    raise ValueError(
+        'No instance type found for the requested configuration: {0:d} CPU '
+        'cores, {1:d} MB memory.'.format(cpu_cores, memory_in_mb))
+
+  def _CreateNetworkInterfaceForVM(self,
+                                   vm_name: str,
+                                   region: Optional[str] = None) -> str:
+    """Create a network interface and returns its ID.
+    This is necessary when creating a VM from the SDK.
+    See https://docs.microsoft.com/en-us/azure/virtual-machines/windows/python
+
+    Args:
+      vm_name (str): The name of the VM to create the network interface for.
+      region (str): Optional. The region in which to create the network
+          interface. Default uses default_region of the AZAccount object.
+
+    Returns:
+      str: The id of the created network interface.
+
+    Raises:
+      ValueError: if vm_name is not provided.
+      RuntimeError: If no network interface could be created.
+    """
+    if not vm_name:
+      raise ValueError(
+          'vm_name must be specified. Provided: {0!s}'.format(vm_name))
+
+    if not region:
+      region = self.default_region
+
+    network_interface_name = '{0:s}-nic'.format(vm_name)
+    ip_config_name = '{0:s}-ipconfig'.format(vm_name)
+
+    # Check if the network interface already exists, and returns its ID if so.
+    try:
+      nic = self.network_client.network_interfaces.get(
+          self.default_resource_group_name, network_interface_name)
+      nic_id = nic.id  # type: str
+      return nic_id
+    except azure_exceptions.CloudError:
+      # NIC doesn't exist, ignore the error and create it
+      pass
+
+    # pylint: disable=unbalanced-tuple-unpacking
+    public_ip, _, subnet = self._CreateNetworkInterfaceElements(
+        vm_name, region=region)
+    # pylint: enable=unbalanced-tuple-unpacking
+
+    creation_data = {
+        'location': region,
+        'ip_configurations': [{
+            'name': ip_config_name,
+            'public_ip_address': public_ip,
+            'subnet': {
+                'id': subnet.id
+            }
+        }]
+    }
+
+    try:
+      request = self.network_client.network_interfaces.create_or_update(
+          self.default_resource_group_name,
+          network_interface_name,
+          creation_data)
+      request.wait()
+    except azure_exceptions.CloudError as exception:
+      raise RuntimeError('Could not create network interface: {0:s}'.format(
+          str(exception)))
+
+    network_interface_id = request.result().id  # type: str
+    return network_interface_id
+
+  def _CreateNetworkInterfaceElements(
+      self,
+      vm_name: str,
+      region: Optional[str] = None) -> Tuple[Any, ...]:
```

Comment:
This is a shortened notation for `Tuple[Any, Any, Any]` since all elements in that tuple are of the same type. You can use the dots `...` whenever you want to specify a type that repeats `n` times

Reply:
Oh I didn't know that - thanks, that's something I just learnt then :)